feature_engine/selection/target_mean_encoding.py

```diff
@@ -0,0 +1,400 @@
+from sklearn.base import TransformerMixin, BaseEstimator
+from sklearn.utils.validation import check_is_fitted
+from sklearn.model_selection import train_test_split
+from sklearn.metrics import roc_auc_score, r2_score
+
+from feature_engine.variable_manipulation import _find_all_variables, _define_variables
+from feature_engine.dataframe_checks import _is_dataframe
+
+import pandas as pd 
+
+class TargetMeanEncoding(BaseEstimator, TransformerMixin):
+    """
+    TargetMeanEncoding
+    ------------------
+        Description
+        -----------
+        Calculates the feature importance.
+
+        For each categorical variable:
+            1) Separate into train and test
+            2) Determine the mean value of the target within each label of the categorical variable using the train set
+            3) Use that mean target value per label as the prediction (using the test set) and calculate the roc-auc.
+
+        For each numerical variable:
+            1) Separate into train and test
+            2) Divide the variable into 100 quantiles
+            3) Calculate the mean target within each quantile using the training set 
+            4) Use that mean target value / bin as the prediction (using the test set) and calculate the roc-auc
+        
+        Implementation
+        --------------
+
+            Private methods
+            ---------------
+                `__target_mean_encoding(self, df_train_, df_test_, var_cols=None)`
+                `__validate_variables(self, X)`
+                    
+            Public methods
+            --------------
+                `fit(self, X, y=None)`
+                `transform(self)`
+                `fit_transform(self, X, y=None)`
+        
+    Parameters
+    ----------
+
+    variables: list, default=None
+        The list of variables to evaluate. If None, the transformer will evaluate all 
+        variables in the dataset associated with the variables_type.
+
+    target: string, default=None
+        The target variable to evaluate. If None, the transformer will return the 
```

Comment:
we do not need to specify target, because this is not a parameter from the init method. Please remove

Reply:
I still not understand your point. Would you mind to share if we don't specify the target in the init method, where should it be initialized in order to pass it to the __target_mean_encoding method?